app/models/spree/order.rb

```diff
@@ -658,7 +659,7 @@ def ensure_available_shipping_rates
     def after_cancel
       shipments.each(&:cancel!)
 
-      OrderMailer.cancel_email(id).deliver_later
+      OrderMailer.cancel_email(id).deliver_later if send_cancellation_email
```

Comment:
I think that, ideally, this doesn't belong to the model but would be handled in controllers. It's a great solution for backwards compatibility though! :+1:

Reply:
You mean the `OrderMailer.cancel_email(id).deliver_later`? 
That belongs to the model transition right? That's why I'v put here the `if send_cancellation_email`
Should we change/correct that before moving to `test ready`?

Reply:
Leave it as it is. It's the easiest and safest way to make this change.

I was just saying that I don't like conditionals in state transitions that depend on data from the controller. The cancellation email is no longer an action that always happens when the order is cancelled. It is now something the admin user controls. That's why I think that it should be called from a controller. The downside is that all other cases in which an order is cancelled, we would also need to send the email. And we could even decide to not send a cancellation email when the user clicks the button because they know they did it. Like you don't get notifications from Github if you did the action yourself. But in this case, it's good to have a digital paper trail of cancellations.

I'm now thinking that it's not good to have this option. There should always be an email. But then we would also need to include all the cancelled items in an email and that's just too much work at the moment. I see this as a workaround.

Anyway, your PR is great. I just don't think that we found the *perfect* solution yet. :wink: